lms/djangoapps/commerce/api/v1/serializers.py

```diff
@@ -25,9 +29,26 @@ class Meta(object):  # pylint: disable=missing-docstring
         fields = ('name', 'currency', 'price', 'sku', 'expires')
 
 
+def validate_course_id(course_id):
+    """
+    Check that course id exists in modulestore.
+    """
+    try:
+        course_key = CourseKey.from_string(unicode(course_id))
+    except InvalidKeyError:
+        raise serializers.ValidationError(
+            "{} is not a valid course key.".format(course_id)
+        )
+
+    if not modulestore().has_course(course_key):
+        raise serializers.ValidationError(
+            "Cannot find course with id {} in the modulestore".format(course_id)
```

Comment:
Make this message less specific to the modulestore, and set it up for translation: `_('Course {} does not exist.')`